pkg/file/validation.go

```diff
@@ -0,0 +1,19 @@
+package file
+
+import "strings"
+
+var InvalidMacOSChars = []rune{':', '\\', '*', '?', '"', '<', '>', '|', 0}
+
+func IsValidMacOSName(fileName string) bool {
+	if fileName == "" {
+		return false
+	}
+
+	for _, char := range InvalidMacOSChars {
+		if strings.ContainsRune(fileName, char) {
+			return false
+		}
+	}
```

Comment:
Nit, but you can use `strings.ContainsAny(fileName, string(InvalidMacOSChars))` instead of the loop (https://pkg.go.dev/strings#ContainsAny).

Reply:
ohhh, that's so cool! https://github.com/fleetdm/fleet/pull/11571/commits/a503a2391bba687b8c578d5371d1c1da82f1df79 beautiful!